src/main/java/org/broadinstitute/hellbender/utils/hmm/segmentation/HiddenStateSegmentRecordReader.java

```diff
@@ -1,53 +1,64 @@
-package org.broadinstitute.hellbender.tools.exome.germlinehmm;
+package org.broadinstitute.hellbender.utils.hmm.segmentation;
 
+import org.broadinstitute.hdf5.Utils;
 import org.broadinstitute.hellbender.tools.exome.SegmentTableColumn;
+import org.broadinstitute.hellbender.tools.exome.Target;
 import org.broadinstitute.hellbender.utils.SimpleInterval;
-import org.broadinstitute.hellbender.utils.hmm.CopyNumberTriState;
 import org.broadinstitute.hellbender.utils.tsv.DataLine;
 import org.broadinstitute.hellbender.utils.tsv.TableReader;
 
 import java.io.File;
 import java.io.IOException;
+import java.util.function.Function;
 
 /**
- * Reads {@link CopyNumberTriStateSegmentRecord} instances from a tab separated table file.
+ * Reads {@link HiddenStateSegmentRecord} instances from a tab separated table file.
  *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
  * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
  */
-public class CopyNumberTriStateSegmentRecordReader extends TableReader<CopyNumberTriStateSegmentRecord> {
+public class HiddenStateSegmentRecordReader<S, T extends Target> extends TableReader<HiddenStateSegmentRecord<S, T>> {
+
+    /**
+     * The inverse call string factory maps a call string to its corresponding hidden state
+     */
+    private final Function<String, S> inverseCallStringFactory;
```

Comment:
"Inverse factory" might be a little too clever, although I understand exactly what you mean.  I would go with something blander like `callStringParser`.